lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -260,23 +273,48 @@ data AddressPool
             )
     } deriving (Generic)
 
-deriving instance (Show (key 'AccountK XPub))
+deriving instance (Show (key 'AccountK XPub), Show (ParentContext chain key))
     => Show (AddressPool chain key)
 
-deriving instance (Eq (key 'AccountK XPub))
+deriving instance (Eq (key 'AccountK XPub), Eq (ParentContext chain key))
     => Eq (AddressPool chain key)
 
-instance (NFData (key 'AccountK XPub))
+instance (NFData (key 'AccountK XPub), NFData (ParentContext chain key))
     => NFData (AddressPool chain key)
 
-instance ((PersistPublicKey (key 'AccountK)), Typeable chain)
+instance PersistPublicKey (key 'AccountK) => Buildable (key 'AccountK XPub) where
+    build key = prefixF 8 xpubF <> "..." <> suffixF 8 xpubF
+      where
+        xpubF = hexF $ serializeXPub key
+
+instance Buildable (ScriptTemplate, Maybe ScriptTemplate) where
+    build (paymentTemplate, delegationTemplateM) =
+        mempty <> " payment script credential: " <> scriptPaymentF <>
+        " delegation script credential: " <> scriptDelegationF
+      where
+        scriptPaymentF = build $ toText paymentTemplate
+        scriptDelegationF = case delegationTemplateM of
+            Just s -> build $ toText s
+            Nothing -> "absent"
+
+instance ToText ScriptTemplate where
```

Comment:
Shouldn't this be a `Buildable` instance? It doesn't look like an appropriate `ToText` instance.